packages/components/src/Autocomplete/Autocomplete.types.ts

```diff
@@ -88,6 +88,16 @@ export interface AutocompleteProps<
     | Array<GenericGetOptionsValue | GenericOption>
     | Promise<Array<GenericGetOptionsValue | GenericOption>>;
 
+  /**
+   * Called by advanced usage scenarios to update the options.
+   * @param newInputText
+   */
+  updateOptions?(
```

Comment:
I wonder if there's a way for us to have our types only include this if you are doing the custom render?

if not, we should communicate this only does something in the `customRenderMenu` flow and won't do anything at all for the default flow

I'd even be tempted to change the name too

`getOptions` and updateOptions` are so similar in name, and if you look behind the curtain even their implementations are almost identical

```
  async function updateSearch() {
    const updatedOptions = await getOptions(inputText);
    const filteredOptions = updatedOptions.filter(option =>
      isOptionGroup(option) ? option.options.length > 0 : true,
    );

    setOptions(mapToOptions(filteredOptions));
  }
```

where we also call `setOptions` on the result of that callback